File: Ontology/DataPropertyRestriction.cs
﻿namespace Ontology
{
    public abstract class DataPropertyRestriction:
        PropertyRestriction,
        IDataPropertyRestriction
    {
        protected readonly IDataPropertyExpression _dataPropertyExpression;

        protected DataPropertyRestriction(
            IDataPropertyExpression dataPropertyExpression
            ) : base(dataPropertyExpression)
        {
            _dataPropertyExpression = dataPropertyExpression;
        }

        IDataPropertyExpression IDataPropertyRestriction.DataPropertyExpression => _dataPropertyExpression;
    }
}

File: Web/ClientApp/src/app/Ontology/DataPropertyRestriction.ts
import { IDataPropertyRestriction } from "./IDataPropertyRestriction";
import { IDataPropertyExpression } from "./IPropertyExpression";
import { PropertyRestriction } from "./PropertyRestriction";

export class DataPropertyRestriction
    extends PropertyRestriction
    implements IDataPropertyRestriction
{
    protected constructor(
        public DataPropertyExpression: IDataPropertyExpression
        )
    {
        super(DataPropertyExpression);
    }
}
